src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -1104,9 +1111,199 @@ static public int beginOfShard(int shardIndex, int shardSize) {
+                if (firstComparison != 0) {
+                    return firstComparison;
+                } else {
+                    // We want true before false
+                    return p2.getRight().compareTo(p1.getRight());
+                }
+            });
+            int numCurrentStarts = 0;
+            int numCurrentEnds = 0;
+            for (int i = 0; i < (breakpoints.size() - 1); i++) {
+                final int currentBreakpoint = breakpoints.get(i).getLeft();
+                final int nextBreakpoint = breakpoints.get(i + 1).getLeft();
+
+                final boolean isCurrentBreakpointStart = breakpoints.get(i).getRight();
+                final boolean isNextBreakpointStart = breakpoints.get(i + 1).getRight();
+
+                // if both breakpoints are starts of intervals, then the result is bp1, bp2-1
+                // if both breakpoints are ends of intervals, then the result is bp1+1, bp2
+                int start = (!isCurrentBreakpointStart && !isNextBreakpointStart ? currentBreakpoint + 1: currentBreakpoint);
+                int end = (isCurrentBreakpointStart && isNextBreakpointStart ? nextBreakpoint - 1 : nextBreakpoint);
+
+                // If the current breakpoint is an end and the next is a start, then we want to shrink both ends.
+                //  Note that this could indicate that we are between intervals for one list of locatables,
+                //   but not the other.
+                final boolean isBetweenIntervals = !isCurrentBreakpointStart && isNextBreakpointStart;
+                if (isBetweenIntervals) {
+                    start++;
+                    end--;
+                }
+
+                // If the next breakpoint is a start and the current is end AND we are not in the middle of an interval,
+                //   then we do NOT add the interval at all (we are between intervals in both lists).
+                if (isCurrentBreakpointStart) {
+                    numCurrentStarts++;
+                } else {
+                    numCurrentEnds++;
+                }
+
+                // We also need to check if start > end, since we could have been between intervals that were one adjacent,
+                //   so we don't want to add an interval for that.
+                if (((!isBetweenIntervals) || (numCurrentStarts > numCurrentEnds)) &&
+                        start <= end){
+                    result.add(new SimpleInterval(contig, start, end));
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Creates a map of which locatables (keys) overlap the other list of locatables (vals)
+     *
+     * Input lists are assumed sorted by interval.  These are lists solely because we need the ordering to make
+     *  this run efficiently.
+     *
+     *  Within a single input list, segments cannot overlap (or duplicate).  If this occurs, behavior is undefined.
```

Comment:
Again, better to validate and fail.  Especially since I do not think you check consistent sorting with respect to contigs in calling code.